### PR TITLE
deprecate created-by annotation for e2e test framework

### LINKED ISSUE #44407: Deprecate created-by annotation, in favor of ControllerRef
TODOs:

* [x] `kubectl describe` list of controllers for an object.
* [x] `kubectl drain` Pod filtering.
* [x] Classifying failed Pods in e2e test framework.
* [x] CronJob grouping Jobs by owner/creator.
* [ ] Announce deprecation of created-by annotation in the release note

@erictune @enisoc @kow3ns @foxish @kargakis @kubernetes/sig-apps-feature-requests

### PR DESCRIPTION
**Pull request by crimsonfaith91**
**What this PR does / why we need it**: This PR deprecates created-by annotation for e2e test framework. This is needed as we now have ControllerRef.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: xref #44407

**Special notes for your reviewer**: This is the third PR for deprecating created-by annotation. Other PRs can be found here: #47469, #47471 

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by janetkuo** [comment]
@k8s-bot ok to test

**Comment by crimsonfaith91** [comment]
@k8s-bot pull-kubernetes-kubemark-e2e-gce test this

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -604,7 +604,7 @@ func WaitForPodsRunningReady(c clientset.Interface, ns string, minPods, allowedN
 				notReady++
 				badPods = append(badPods, pod)
 			default:
-				if _, ok := pod.Annotations[v1.CreatedByAnnotation]; !ok {
+				if controller.GetControllerOf(pod.GetObjectMeta()) == nil {
```
Comment (enisoc) [inline test/e2e/framework/util.go]: Usually it should work to just do `GetControllerOf(pod)`.

Comment (crimsonfaith91) [inline test/e2e/framework/util.go]: `pod` is of type `"k8s.io/kubernetes/pkg/api/v1".Pod`, but `controller.GetControllerOf()` requires `k8s.io/apimachinery/pkg/apis/meta/v1".Object`.
`GetObjectMeta()` is to ensure the types match.

Comment (janetkuo) [inline test/e2e/framework/util.go]: Just use `controller.GetControllerOf(&pod)`

Comment (crimsonfaith91) [inline test/e2e/framework/util.go]: thanks! will send out another commit soon. :)

**Comment by crimsonfaith91** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by crimsonfaith91** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by crimsonfaith91** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by crimsonfaith91** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Review by janetkuo — APPROVED** [review]

**Review by enisoc — APPROVED** [review]
LGTM

**Comment by soltysh** [comment]
lgtm

**Pull request merged**